### DISCUSSION TITLE
Behavior for overlaping matchFileNames in packageRules

### DISCUSSION THREAD

**Discussion by Nihileon** [renovatebot/renovate#33210] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

github

### Please tell us more about your question or problem

Hi, I have some overlaps for the packageRules as below, what would be the behavior of `"^services/exporter/.*\\.ya?ml$"`? Does it use the first rule, or is it overridden by the second rule? Thank you!

```
  packageRules: [
    {
      matchFileNames: ["^services/**"],
      additionalBranchPrefix: "test-",
      commitMessageSuffix: "(test)"
    },
    {
      matchFileNames: ["^services/exporter/.*\\.ya?ml$"],
      additionalBranchPrefix: "services-",
      commitMessageSuffix: "(services)",
      schedule: ["0 0 10 * *"]
    }
  ]
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
The settings from last matched rule will override previous ones if same settings are configured.

**Reply by Nihileon** [reply]
> The settings from last matched rule will override previous ones if same settings are configured. 

thank you for your answer!
